### PR TITLE
Autocomplete and changed the way opener files are loaded to use a folder instead

### PR DESCRIPTION
**Pull request by Hikari-Haru**
Autocomplete for settings makes it easier to add more settings in the future as they're always updated. It also gives you a returned list based on what you've typed in so far.

I've also added autocomplete values for some settings parameters. It can get a little confused though if one goes back and changes the parameter to something else then back to the value, I don't quite understand it but there's probably some caching. For any values that aren't str I had to defer and let it not get any results, so you could still input, since autocomplete is typically only for str inputs.

Made opener files a separate option so that you can combine an opener file with a separate prompt after it. `opener_file + opener` also forces the minimal starter text

Moved the files over into their own folder because it was kinda messy and you could load `usage.txt` as an opener lol
Don't know if there are any docker changes necessary as I don't use it

There's an autocomplete for `opener_file` which gives you a list of files in the directory to choose from

Added some example opener files from [awesome-chatgpt-prompts](https://github.com/f/awesome-chatgpt-prompts)

Example of the settings value autocomplete
image

### CODE REVIEW

**Comment by Kav-K** [comment]
Opening from an opener file doesn't seem to add the custom conversation start into the conversation, I tried opening with the javascript console file and I got:

image

**Comment by Kav-K** [comment]
Apart from that, everything else looks good

File: cogs/gpt_3_commands_and_converser.py [hunk 1]
```diff
@@ -781,22 +795,24 @@ async def converse(
...
 
-        if not opener:
+        if not opener or not opener_file:
             user_id_normalized = user.id
         else:
             user_id_normalized = ctx.author.id
             # Pre-check for opener, check if they provided a valid file if it is indeed a file.
             # If the opener ends in .txt, its a file and we want to load it
-            if opener.endswith(".txt"):
+            if opener_file.endswith(".txt"):
```
Comment (Kav-K) [inline cogs/gpt_3_commands_and_converser.py:804]: see my other comments above :)

**Review by Kav-K — CHANGES_REQUESTED** [review]

**Comment by Hikari-Haru** [comment]
Got them lovely conditionals messes up

File: models/autocomplete_model.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        }
+        if ctx.options["parameter"] in values.keys():
+            return[value for value in values[ctx.options["parameter"]]]
+        else:
+            await ctx.interaction.response.defer() # defer so the autocomplete in int values doesn't error but rather just says not found
+            return []
+
+class File_autocompleter:
+    async def get_openers(ctx: discord.AutocompleteContext):
+        return [file for file in os.listdir('openers') if file.startswith(ctx.value.lower())][:25] # returns the 25 first files from your current input
```
Comment (Kav-K) [inline models/autocomplete_model.py:34]: Can we add some error handling to this? For example if the openers directory is accidentally deleted by the user or something, maybe we should pop an error and say that there is no `openers` directory.

(when autocomplete happens with no openers directory)
```
future: <Task finished name='Task-36' coro=<ApplicationCommandMixin.on_application_command_auto_complete.<locals>.callback() done, defined at /usr/local/lib/python3.9/dist-packages/discord/bot.py:853> exception=FileNotFoundError(2, 'No such file or directory')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/discord/bot.py", line 856, in callback
    return await command.invoke_autocomplete_callback(ctx)
  File "/usr/local/lib/python3.9/dist-packages/discord/commands/core.py", line 1316, in invoke_autocomplete_callback
    await command.invoke_autocomplete_callback(ctx)
  File "/usr/local/lib/python3.9/dist-packages/discord/commands/core.py", line 1003, in invoke_autocomplete_callback
    result = await result
  File "/home/gptbot/models/autocomplete_model.py", line 34, in get_openers
    return [file for file in os.listdir('openers') if file.startswith(ctx.value.lower())][:25] # returns the 25 first files from your current input
FileNotFoundError: [Errno 2] No such file or directory: 'openers'
```

**Review by Kav-K — CHANGES_REQUESTED** [review]

File: models/autocomplete_model.py (hunk 2 above)
Comment (Hikari-Haru) [reply models/autocomplete_model.py:34]: Yeah no problem

**Comment by Kav-K** [comment]
Looks good, I also love the new prompts and that prompt repo!

**Pull request merged**